addons/account/tests/test_reconciliation_matching_rules.py

```diff
@@ -296,3 +303,47 @@ def test_auto_reconcile(self):
             {'partner_id': self.partner_2.id, 'debit': 0.0, 'credit': 1000.0},
             {'partner_id': self.partner_2.id, 'debit': 1000.0, 'credit': 0.0},
         ])
+
+    def test_reverted_move_matching(self):
+        AccountMove = self.env['account.move']
+        move = AccountMove.create({
+            'name': 'To Revert',
+            'journal_id': self.bank_journal.id,
+        })
+
+        partner = self.env['res.partner'].create({'name': 'Eugene'})
+        AccountMoveLine = self.env['account.move.line'].with_context(check_move_validity=False)
+        payment_payable_line = AccountMoveLine.create({
+            'account_id': self.account_pay.id,
+            'move_id': move.id,
+            'partner_id': partner.id,
+            'name': 'One of these days',
+            'debit': 10,
+        })
+        payment_bnk_line = AccountMoveLine.create({
```

Comment:
@kebeclibre I highly encourage you to define your one2many like:
```
'line_ids': [(0, 0, {...}), (0, 0, {...}), ....]
```

...because it's:
- more efficient (create batch)
- more readable (less code)
- avoid trick like `.with_context(check_move_validity=False)`

Reply:
I feel you bro, but in tests we usually need a reference to a specific line to check it against some values,
just like I do at l347
Otherwise, we would have to filter.map with lambdas, which is, I guess, not very clearer :D